backend/src/test/java/de/bund/digitalservice/ris/caselaw/adapter/DatabasePatchMapperServiceTest.java

```diff
@@ -292,4 +294,160 @@ void extractAndStoreBase64Images_withInvalidHtml_shouldKeepOriginalText() {
     String resultHtml = ((PathValueOperation) result.getOperations().get(0)).getValue().textValue();
     assertEquals(invalidHtml, resultHtml);
   }
+
+  @Test
+  void givenPatchWithNoChanges_whenComparingToStoredText_thenRemoveOperation() {
+    // given
+    var textWithTextCheckTags = "<p>Some text with some flag</p>";
+    var patchPath = "/longTexts/tenor";
+    JsonPatch patch =
+        new JsonPatch(
+            List.of(new ReplaceOperation(patchPath, new TextNode(textWithTextCheckTags))));
+    var expectedStoredText = "<p>Some text with some flag</p>";
+    var docUnit =
+        Decision.builder()
+            .uuid(UUID.randomUUID())
+            .documentNumber("YYDocNumber")
+            .longTexts(LongTexts.builder().tenor(expectedStoredText).build())
+            .build();
+
+    var expectedPatch = new JsonPatch(Collections.emptyList());
+
+    // when
+    var result = service.removeOpsWhereContentNotChanged(patch, docUnit);
+
+    // then
+    assertEquals(expectedPatch.getOperations(), result.getOperations());
+  }
+
+  @Test
+  void
+      givenPatchWithMultipleOperations_whenNoLongTextChangesFound_thenVerifyOnlyLongTextReplaceOperationSkipped()
+          throws JsonProcessingException {
+    // given
+    ObjectMapper mapper = new ObjectMapper();
+
+    String valueJson =
+        "{"
+            + "\"id\":\"65693d24-30ab-4e8b-b82a-112954b8c09b\","
+            + "\"hasChildren\":true,"
+            + "\"identifier\":\"AR-01\","
+            + "\"text\":\"Arbeitsvertrag: Abschluss, Klauseln, Arten, Betriebsübergang\","
+            + "\"linkedFields\":[],"
+            + "\"norms\":["
+            + "  {\"abbreviation\":\"BGB\",\"singleNormDescription\":\"§ 611a\"},"
+            + "  {\"abbreviation\":\"GewO\",\"singleNormDescription\":\"§ 105\"}"
+            + "],"
+            + "\"children\":[],"
+            + "\"parent\":{"
+            + "  \"id\":\"ea9f6ee0-9457-4730-9c9b-c30127fbab29\","
+            + "  \"hasChildren\":true,"
+            + "  \"identifier\":\"AR\","
+            + "  \"text\":\"Arbeitsrecht\","
+            + "  \"linkedFields\":[],"
+            + "  \"norms\":null,"
+            + "  \"children\":[],"
+            + "  \"parent\":null"
+            + "}"
+            + "}";
+
+    var valueNode = mapper.readTree(valueJson);
+    JsonPatch patch =
+        new JsonPatch(
+            List.of(new AddOperation("/contentRelatedIndexing/fieldsOfLaw/0", valueNode)));
+
+    DocumentationUnit docUnit =
+        Decision.builder().uuid(UUID.randomUUID()).documentNumber("YYDocNumber").build();
+
+    // when
+    JsonPatch result = service.removeOpsWhereContentNotChanged(patch, docUnit);
+
+    // then
+    assertEquals(patch.getOperations(), result.getOperations());
+  }
+
+  @Test
+  void givenMultipleOperations_whenOnlyLongTextIsUnchanged_thenOnlyLongTextReplaceIsRemoved()
+      throws Exception {
+    // given
+    ObjectMapper mapper = new ObjectMapper();
+
+    String valueJson =
```

Comment:
for the next test. you should use fields of law as patch. because it is one of the complex object. filenumbers are much easiert or any single value field.